src/test/scheduler_tests.cpp

```diff
@@ -171,8 +172,6 @@ BOOST_AUTO_TEST_CASE(mockforward)
     size_t num_tasks = scheduler.getQueueInfo(first, last);
     BOOST_CHECK_EQUAL(num_tasks, 3ul);
 
-    std::thread scheduler_thread([&]() { scheduler.serviceQueue(); });
-
     // bump the scheduler forward 5 minutes
     scheduler.MockForward(boost::chrono::seconds(5*60));
```

Comment:
It looks like this line (or any line after it, but before the next BOOST_CHECK) still fails. E.g.

```
Running 3 test cases...

Test cases order is shuffled using seed: 902263820

Entering test module "Bitcoin Core Test Suite"

test/scheduler_tests.cpp(11): Entering test suite "scheduler_tests"

test/scheduler_tests.cpp(156): Entering test case "mockforward"

terminate called after throwing an instance of 'boost::wrapexcept<boost::condition_error>'

  what():  boost::condition_variable::do_wait_until failed in pthread_cond_timedwait: Invalid argument

unknown location(0): fatal error: in "scheduler_tests/mockforward": signal: SIGABRT (application abort requested)

test/scheduler_tests.cpp(173): last checkpoint

test/scheduler_tests.cpp(156): Leaving test case "mockforward"; testing time: 430us

test/scheduler_tests.cpp(112): Entering test case "singlethreadedscheduler_ordered"

test/scheduler_tests.cpp(112): Leaving test case "singlethreadedscheduler_ordered"; testing time: 7233us

test/scheduler_tests.cpp(38): Entering test case "manythreads"

test/scheduler_tests.cpp(38): Leaving test case "manythreads"; testing time: 8082us

test/scheduler_tests.cpp(11): Leaving test suite "scheduler_tests"; testing time: 15837us

Leaving test module "Bitcoin Core Test Suite"; testing time: 15956us